### PR TITLE
VMware: New module: vmware_tag

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This module is based on vSphere REST API. This module allows
user to manage various tags and their association with
categories. This fix also adds vCenter REST client library which can
be re-used for other REST based modules.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/module_utils/vmware_rest_client.py
lib/ansible/modules/cloud/vmware/vmware_tag.py
lib/ansible/utils/module_docs_fragments/vmware_rest_client.py

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

**Comment by angystardust** [comment]
Great module, @Akasurde! I'm wondering if you also plan to have a `vmware_tag_facts` module... :thinking:

**Comment by Akasurde** [comment]
@angystardust Yes. Once the `vmware_rest_client.py` is merged. There are bunch of other modules following it and `vmware_tag_facts` is one of them.

File: lib/ansible/modules/cloud/vmware/vmware_tag.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  vmware_tag:
+    hostname: 10.65.223.91
+    username: administrator@vsphere.local
+    password: Esxi@123$
+    validate_certs: False
+    tag_name: Sample_Tag_0002
+    tag_description: Some fancy description
+    state: present
+
+- name: Update tag description
```
Comment (angystardust) [inline lib/ansible/modules/cloud/vmware/vmware_tag.py]: This should be "Delete a tag"

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_tag.py]: Done.

**Comment by Akasurde** [comment]
@angystardust @pdellaert @dericcrago Can you please review this ?

**Review by pdellaert — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
Closing and re-opening for CI trigger.

**Pull request merged**